executor/runtime/docker/docker.go

```diff
@@ -1640,6 +1649,137 @@ func (r *DockerRuntime) Start(parentCtx context.Context, pod *v1.Pod) (string, *
 	return logDir, details, statusMessageChan, err
 }
 
+func (r *DockerRuntime) reportContainerStatusMetrics(ctx context.Context) {
+	l := log.WithField("taskID", r.c.TaskID())
+	defer func() {
+		if r := recover(); r != nil {
+			l.Warnf("Unexpected panic when reporting container status metrics: %#v", r)
+		}
+	}()
+	spectatordSocket := fmt.Sprintf("/var/lib/titus-inits/%s/root/run/spectatord/spectatord.unix", r.c.TaskID())
+	conn, err := connectToSpectatord(spectatordSocket, 3) // Retry 3 times before giving up.
+	if err != nil {
+		l.WithError(err).Warnf("Dial error with spectatord socket %s", spectatordSocket)
+		return
+	}
+	defer conn.Close()
+	reportTicker := time.NewTicker(1 * time.Minute)
+	defer reportTicker.Stop()
+	for {
+		select {
+		case <-ctx.Done():
+			l.WithError(ctx.Err()).Debug("Stopping reporting container status metric")
+			return
+		case <-reportTicker.C:
+			metricLines := r.containerStatusMetrics()
+			for _, m := range metricLines {
+				if _, err := conn.Write([]byte(m)); err != nil {
+					l.WithError(err).Warnf("Error sending metric to spectatord, metric string: %s", m)
+				}
+			}
+		}
+	}
+}
+
+func connectToSpectatord(spectatordSocket string, retriesLeft int) (net.Conn, error) {
+	conn, err := net.Dial("unixgram", spectatordSocket)
+	if err != nil {
+		if retriesLeft--; retriesLeft > 0 {
+			time.Sleep(1 * time.Second)
+			return connectToSpectatord(spectatordSocket, retriesLeft)
+		}
+	}
+	return conn, err
+}
+
+func (r *DockerRuntime) containerStatusMetrics() []string {
+	l := log.WithField("taskID", r.c.TaskID())
+	var metricLines []string
+	// 1. Loop through all containers.
+	for _, c := range r.c.Pod().Status.ContainerStatuses {
+		// 2. Get container metadata and state.
+		metricTags := make(map[string]string)
+		metricTags["nf.container"] = c.Name
+		metricTags["nf.process"] = processNameForContainer(c.Name, r.c.Pod())
+		imageName, imageVersion, err := imageNameAndDigest(c.Image, c.Name)
+		if err != nil {
+			l.WithError(err).Warnf("Error getting image metric tags for container status metrics")
```

Comment:
We should `continue` here. If we got any error here, our line will be bogus and not worth sending

Reply:
If the image string is bad, does it mean there's a bigger problem? Is it worth reporting an unhealthy status in this case?

Reply:
The string couldn't be so bad that we got this far (that the container launched)
Other errors will crop up.

Reply:
Makes sense. PR is updated with `continue` here, so it's ready to go :)